### PR TITLE
Bundle libtpu with reactant by downloading it from google

### PR DESCRIPTION
**Pull request by gbaraldi**
Google doesn't bundle a license with this so we need to find it somewhere. 
It used to exist in pypi https://pypi.org/project/libtpu-nightly/ as MIT but for reasons it got yanked.

### CODE REVIEW

File: R/Reactant/build_tarballs.jl [hunk 1]
```diff
@@ -206,13 +208,19 @@ if [[ "${bb_full_target}" == *darwin* ]]; then
 fi
 if [[ "${bb_full_target}" == *mingw* ]]; then
     mv ${libdir}/libReactantExtra.so ${libdir}/libReactantExtra.dll
+
+if [[ $bb_full_target == x86_64-linux-gnu ]]; then
+    cd $WORKSPACE/srcdir
+    unzip -d onnxruntime-$target-cuda libtpu.whl
+    mv $WORKSPACE/srcdir/libtpu.so ${libdir}/libtpu.so
```
Comment (giordano) [inline R/Reactant/build_tarballs.jl:215]:
```suggestion
    install -Dvm 755 $WORKSPACE/srcdir/libtpu.so -t ${libdir}
```

**Comment by vchuravy** [comment]
So `pytorch/xla` is not redistributing libtpu. https://github.com/pytorch/xla?tab=readme-ov-file#tpu

Instead one has to do: `pip install torch~=2.4.0 torch_xla[tpu]~=2.4.0 -f https://storage.googleapis.com/libtpu-releases/index.html`

**Comment by vchuravy** [comment]
Maybe we could just make it an Artifact? https://pkgdocs.julialang.org/v1/artifacts/

File: R/Reactant/build_tarballs.jl [hunk 2]
```diff
@@ -206,13 +208,19 @@ if [[ "${bb_full_target}" == *darwin* ]]; then
 fi
 if [[ "${bb_full_target}" == *mingw* ]]; then
     mv ${libdir}/libReactantExtra.so ${libdir}/libReactantExtra.dll
+
+if [[ $bb_full_target == x86_64-linux-gnu ]]; then
+    cd $WORKSPACE/srcdir
+    unzip -d onnxruntime-$target-cuda libtpu.whl
```
Comment (wsmoses) [inline R/Reactant/build_tarballs.jl:214]: this probably shouldn't be onnx runtime

**Comment by wsmoses** [comment]
Superceeded by https://github.com/EnzymeAD/Reactant.jl/pull/93

**Pull request closed without merging**